### DISCUSSION TITLE
SVG grouping bug -  coordinate transformation in group distoring path coordinates of neighboring group

### DISCUSSION THREAD

**Discussion by TIEeis** [clientIO/joint#2682] [Q&A]
### Introduction

I am using some shapes.bpmn2 in a process designer (short video attached) where moving a scroll bar in a menu triggers some distortion on elements close by, specifically the link-elements which attach between elements that function as nodes.

Both elements are in their respective groups, siblings one level deeper
If this is a known issue, please link to it and close this discussion.

https://github.com/clientIO/joint/assets/149580666/240ad1fa-8aee-4165-954f-41156b307be8

### Steps to reproduce

Please find video with DevTool clip and scrollbar movement attached

### Restrictions & Constraints

jointJS+ v3.6.1 a solution within this version would be great

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment]
You might need to use the [magnetSelector](https://docs.jointjs.com/api/dia/attributes/#magnetselector) to tell the link to connect to the body `<rect/>` instead of the shape wrapping `<g/>`.

```ts
el.attr('root/magnetSelector', 'body');
```

The `clip-path` (that is used to implement the scrolling) visually clips the shape, but it does not modify its bounding box. When you scroll the content, the invisible parts may still affect the wrapping `<g/>` bounding box which can lead for the link anchor to change its position.

**Reply by TIEeis** [reply]
It appears the issue is a combination of using the connection point 'defaultAnchor' with name 'perpendicular', while connecting the link on the contents (the list) of the body, instead of the body.

Thanks a lot!

**Reply by TIEeis** [reply]
Hi Roman, following up on this bug issue - I was able to find one of your own Demos that is also demonstrating the behavior I described

https://www.jointjs.com/demos/data-mapping

this is the only demo where a headeredRecord with a big enough list of entries to create a scrollbar is shown, I can see here, that the arrows move about as well - indicating that they point inside the element; not to the element.

Is there anything which can remedy this, a setting I am missing?

**Reply by kumilingus** [reply]
This is different. In this case, it is the links connecting the record items, not the item as a whole. The links track the position of the items. When an item is hidden, the link is pinned to the top/bottom of the record.
